tfidf_vectorizer/extract_keywords_tfidf_scratch.py

```diff
@@ -0,0 +1,187 @@
+import math
+import string
+import sys
+
+import nltk
+from nltk.tokenize import wordpunct_tokenize
+
+
+class TF_IDF_Scratch:
+    """
+    Initializes the algorithm with preferred language,
+    a list of stopwords, and a list of punctuations
+
+    If language is unspecified, it defaults to English
+    If stopwords is unspecified, it defaults to the list of stopwords provided by NLTK
+    If punctuations is unspecified, it defaults to Python's string.punctuation
+    """
+
+    def __init__(self, language="english", stopwords=None, punctuations=None):
+        # Using the NLTK stopwords if not specified
+        self.stopwords = stopwords
+        if self.stopwords is None:
+            self.stopwords = nltk.corpus.stopwords.words(language)
+
+        # Using Python's string.punctuation if not specified
+        self.punctuations = punctuations
+        if self.punctuations is None:
+            self.punctuations = string.punctuation
+
+        # Building the ignored set by joining the stopwords and punctuations list
+        # Constructing a set to remove duplicates and fast lookup
+        self.ignored = set(self.stopwords + list(self.punctuations))
+
+        # Member attributes
+        self.num_docs = 0
+        self.word_freq_docs = {}
+        self.tf_idf = {}
+        self.tf = {}
+        self.idf = {}
+
+    """
+    Cleans a document by removing stopwords and punctuation
+
+    Returns a string which represents the cleaned document
+    """
+
+    def clean_document(self, doc):
+        # Tokenizing and converting to lowercase
+        tokens = wordpunct_tokenize(doc)
+        cleaned_doc = " ".join([
+            token.lower() for token in tokens
+            if not token.lower() in self.ignored
+        ])
+        return cleaned_doc
+
+    """
+    Digests a single document and counts the unique words that appear in it
+    """
+
+    def digest_doc(self, doc):
+        cleaned_doc = self.clean_document(doc)
+        for word in set(cleaned_doc.split()):
+            if word in self.word_freq_docs:
+                self.word_freq_docs[word] += 1
+            else:
+                self.word_freq_docs[word] = 1
+
+        self.num_docs += 1
+
+    """
+    Builds a frequency counter of words from a corpus of docs
+    """
+
+    def digest_docs(self, docs):
```

Comment:
@2bit-hack could you modify the name of this function ? It is too similar to the above one, and would create confusion, even though you've described it.
What about something like ```word_freq_docs``` ? Or something different from ```digest_doc```.
And also update the name of this function wherever you have used it next.

Reply:
How about `build_word_freq_docs`?